src/gc-arena/src/types.rs

```diff
@@ -48,10 +61,41 @@ impl GcFlags {
         self.0.get() & 0x4 != 0x0
     }
 
+    /// This is `true` if we've traced a weak pointer during to this `GcBox`
+    /// during the most recent `Phase::Propagate`. This is reset back to
+    /// `false` during `Phase::Sweep`.
+    #[inline]
+    pub(crate) fn has_weak_ref(&self) -> bool {
+        self.0.get() & 0x8 != 0x0
+    }
+
+    /// Determines whether or not we've dropped the stored `dyn Collect` value.
+    /// When we garbage-collect a `GcBox` that still has outstanding weak pointers,
+    /// we set `alive` to false. When there are no more weak pointers remaining,
+    /// we will deallocate the `GcBox`, but skip dropping the `dyn Collect` value
+    /// (since we've already done it).
+    #[inline]
+    pub(crate) fn alive(&self) -> bool {
```

Comment:
Can this just be `is_live` or `is_dropped`?

Reply:
I've renamed this to `is_alive`.